product_pricelist_supplierinfo_coefficient/models/product_pricelist_item.py

```diff
@@ -0,0 +1,32 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import fields, models
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    coefficient = fields.Float(string='Coefficient', default=0.0)
+
+    def _compute_price(self, product, quantity, uom, date, currency=None):
+        result = super()._compute_price(product, quantity, uom, date, currency)
```

Comment:
According to `_compute_price` original definition currency is an optional parameter. For those parameters, we must ensure passing parent parameter:
```suggestion
        result = super()._compute_price(product, quantity, uom, date, currency=currency)
```